lib/python/treadmill_aws/ec2client.py

```diff
@@ -189,6 +206,169 @@ def get_vpc_id_by_tags(ec2_conn, tags):
     return get_vpc_by_tags(ec2_conn, tags)['VpcId']
 
 
+def create_image(
+        ec2_conn, image_name, cloud_init, base_image_id,
+        image_instance_type, image_instance_key, image_instance_role,
+        image_instance_secgroup_ids, image_subnet_id, image_version
+):
+    # pylint: disable=R0915
+    """Creates AWS AMI."""
+
+    instance = create_instance(
+        ec2_conn, image_name, cloud_init, base_image_id,
+        image_instance_type, image_instance_key, image_instance_role,
+        image_instance_secgroup_ids, image_subnet_id, image_version
+    )
+
+    instance_id = instance[0]['InstanceId']
+
+    attempt = 0
+
+    while True:
+        try:
+
+            instance_status = get_instance_by_id(
+                ec2_conn, instance_id
+            ).get('State')
+
+            assert len(instance_status) == 2, \
+                'Waiting AWS to update EC2 instance status'
+
+            attempt = 0
+
+            break
+
+        except (ClientError, AssertionError) as error:
+            assert attempt <= 12, \
```

Comment:
certainly no asserts for error handling. 

log - use _LOGGER.info/warn/error/critical.

_LOGGER is create top level if not there already, _LOGGER = logging.getLogger(__name__)